Framework/API/src/AlgorithmManager.cpp

```diff
@@ -73,6 +71,36 @@ IAlgorithm_sptr AlgorithmManagerImpl::create(const std::string &algName, const i
   return alg;
 }
 
+/** Creates and initialises an instance of an algorithm.
+ *
+ * The algorithm gets tracked in the list of "managed" algorithms,
+ * which is shown in GUI for cancelling, etc.
+ *
+ * @param  algName :: The name of the algorithm required
+ * @param  version :: The version of the algorithm required, if not defined most
+ * recent version is used -> version =-1
+ * @return A pointer to the created algorithm
+ * @throw  NotFoundError Thrown if algorithm requested is not registered
+ * @throw  std::runtime_error Thrown if properties string is ill-formed
+ * This function is the same as the create function but doesn't remove finished algorithms
+ * It has been introduced because removing algorithms requires the GIL to be held
+ */
+IAlgorithm_sptr AlgorithmManagerImpl::createFromPython(const std::string &algName, const int &version) {
```

Comment:
Would it be possible to use this new function in the normal `AlgorithmManagerImpl::create` method to minimize the code reuse? For example we could change the  `AlgorithmManagerImpl::create` method to use this new GIL-safe method:

e.g.
```
IAlgorithm_sptr AlgorithmManagerImpl::create(const std::string &algName, const int &version) {
  removeFinishedAlgorithms();
  return createGILSafe(algName, version);
}
```

Reply:
I was trying to minimize the changes but yeah it will be better to refactor it